core/schema/service.go

```diff
@@ -101,16 +102,14 @@ func (s *serviceSchema) start(ctx context.Context, parent dagql.Instance[*core.S
 	return dagql.NewID[*core.Service](parent.ID()), nil
 }
 
-func (s *serviceSchema) stop(ctx context.Context, parent dagql.Instance[*core.Service], args struct{}) (core.ServiceID, error) {
-	if err := parent.Self.Stop(ctx, parent.ID()); err != nil {
-		return core.ServiceID{}, err
-	}
+type serviceStopArgs struct {
+	Kill dagql.Optional[dagql.Boolean]
```

Comment:
You can also do `*bool` here:

```suggestion
	Kill *bool
```

Setting a default value should also do what you want:

```suggestion
	Kill bool `default:"false"`
```

Reply:
Done, thanks!

Spent some time trying to make sure that it was applied as an options struct, instead of a required arg.